src/lib/Imp.hs

```diff
@@ -137,10 +137,10 @@ toImpExportedFunction cc lam@(NaryLamExpr (NonEmptyNest fb tb) effs body) (Abs b
   let (ccFormals, ccCtx) = ccPrepareFormals cc baseArgBs ptrFormals
   dropSubst $ buildImpFunction CEntryFun ccFormals \ccActuals -> do
     (args, ptrs)   <- ccUnpackActuals ccCtx ccActuals
-    resDestAbsPtrs <- applyNaryAbs (sink resDestAbsArgsPtrs) args
-    resDest        <- applyNaryAbs resDestAbsPtrs            ptrs
     argAtoms <- extendSubst (baseArgBs @@> map SubstVal (Var <$> args)) $
       traverse (translateBlock Nothing) $ fromListE argRecons
+    resDestAbsPtrs <- applyNaryAbs (sink resDestAbsArgsPtrs) (map SubstVal argAtoms)
+    resDest        <- applyNaryAbs resDestAbsPtrs            ptrs
```

Comment:
Is this change meaningful in any way, or just moving the code around?

Reply:
Actually it's important! If you substitute before applying the recon then the substitution isn't type-preserving. It took me a while to find that bug. I'll write a comment.

Reply:
Ok I didn't notice the change from `args` to `argRecons`. This is why I asked :) Thanks for explaining!